src/plugins/map/MapObject.cpp

```diff
@@ -1260,67 +1281,99 @@ void MapObjectMapLoc::UpdateText()
 
 //----------------------------------------------------------------------------
 
-MapObjectMapLoc* GetMapLocByIndex(size_t index)
+MapObjectMapLoc* MapLocTemplate::GetMapLoc()
 {
-	--index;
-
-	if (index >= sMapLocs.size())
-		return nullptr;
-
-	return sMapLocs[index];
+	return m_MapLoc;
 }
 
-MapObjectMapLoc* GetMapLocByTag(const std::string& tag)
+MapLocTemplate* GetMapLocByTag(const std::string& tag)
```

Comment:
With the correction to string comparison, GetMapLocByTag can be changed to take a `string_view` which will make it work with things that aren't std::string.